app/models/nic/managed.rb

```diff
@@ -42,6 +41,22 @@ def queue_with_host
     end
     alias_method_chain :queue, :host
 
+    def progress_report_id
+      if host && host.respond_to?(:progress_report_id)
+        host.progress_report_id
+      else
+        @progress_report_id ||= Foreman.uuid
```

Comment:
Use `super` here (if it'll work) or use `alias_method_chain :progress_report_id, :host` and `progress_report_id_without_host` in the same way as the `queue` code above - then you don't have to copy/paste the contents of Orchestration::ProgressReport. The original methods are already included in this class because it includes Orchestration, so use them.